docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,271 @@ keywords:
 
-## Create an extension point
+:::
 
-Use the `getPluginLinkExtensions` method in `@grafana/runtime` to create an extension point within your plugin.
+## Create an extension point
 
 :::danger
 
 When you create an extension point in a plugin, you create a public interface for other plugins to interact with. Changes to the extension point ID or its context could break any plugin that attempts to register a link inside your plugin.
 
 :::
 
-The `getPluginLinkExtensions` method takes an object consisting of the `extensionPointId`, which must begin `plugins/<PLUGIN_ID>`, and any contextual information that you want to provide. The `getPluginLinkExtensions` method returns a list of extension links that your program can then loop over.
+You can easily create an extension point using the following functions (they live in `@grafana/runtime`) to fetch extensions for a certain extension point ID:
+
+### `usePluginExtensions()`
+
+:::note
+
+**Performance tip:** in case the `context` object is created dynamically, make sure to wrap it into a `useMemo()` to prevent unnecesssary rerenders. [More info](#)
+
+:::
+
+The `usePluginExtensions` react hooks returns the list of extensions that are registered for a certain extension point ID. The hook will dynamically update its return value when the list of extensions changes - this usually happens when extensions are registered during runtime due to dynamic plugin loading.
+
+#### Syntax
+
+```tsx
+usePluginExtensions(options);
+usePluginLinkExtensions(options); // Only returns extensions that have type `type="link"`
+usePluginComponentExtensions(options); // Only returns extensions that have type `type="component"`
+```
+
+#### Parameters
+
+##### `options.extensionPointId` - _string_
+
+The unique identifier of your extension point. It must begin with `plugins/<PLUGIN_ID>`, for example: `plugins/myorg-super-app`.
+
+##### `options?.context` - _object (Optional)_
+
+An arbitrary object, that contains information related to your extension point which you would like to share with the extensions, for example: `{ baseUrl: '/foo/bar' }`. This parameter is not available for component extensions, for those you can pass contextual information using the component props. **Note:** the provided context object always gets frozen (turned immutable) before being shared with the extensions.
+
+##### `options?.limitPerPlugin` - _number (Optional)_
+
+It can be used to limit maximum how many extensions should be returned from the same plugin. It can be useful in cases when there is limited space on the UI to display extensions.
```

Comment:
```suggestion
Use thie parameter to set the maximum value for how many extensions should be returned from the same plugin. It can be useful in cases when there is limited space on the UI to display extensions.
```